client/Utility/Toolbox.ts

```diff
@@ -25,6 +25,20 @@ export function probablyUniqueString(): string {
         let index = Math.floor(Math.random() * chars.length);
         probablyUniqueString += chars[index];
     }
-    
+
     return probablyUniqueString;
-}
\ No newline at end of file
+}
+
+export function combatantCountsByName<T>(name: string, counts: T, oldName?: string): T {
```

Comment:
This method need not be generically typed. I think `counts` is a `{ [name: string]: number }`, it would make sense to also add that as a type annotation where the field is declared in Encounter.ts

Reply:
I wasn't sure if the fact that it was also a KnockoutObservable needed to be accounted for, and I wasn't sure how to define that as a type annotation. I'll give it a try with the native type declaration.

Reply:
Observables are generically typed. So ko.observable("foo") is a KnockoutObservable<string>. It looks like Encounter.ts actually already has this field typed so you should be good.